code/game/gamemodes/miniantags/morph/morph_event.dm

```diff
@@ -17,12 +17,16 @@
 
 		var/datum/mind/player_mind = new /datum/mind(key_of_morph)
 		player_mind.active = TRUE
-		if(!GLOB.xeno_spawn)
-			kill()
-			return
-		var/mob/living/simple_animal/hostile/morph/S = new /mob/living/simple_animal/hostile/morph(pick(GLOB.xeno_spawn))
+		var/list/vents = get_valid_vent_spawns(exclude_mobs_nearby = TRUE)
+		if(!length(vents))
+			message_admins("Warning: No suitable vents detected for spawning morphs. Force picking from station vents regardless of state!")
+			vents = get_valid_vent_spawns(unwelded_only = FALSE, min_network_size = 0)
+		var/obj/vent = pick(vents)
```

Comment:
Might want a length check on this one too? Picking an empty list will runtime. Will probably only be an actual problem on test_tiny, but still worth adding a sanity check for.